src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationDiagnosticsCollector.java

```diff
@@ -35,40 +35,34 @@ protected String getDiagnosticSource() {
     }
 
     public void collectDiagnostics(PsiJavaFile unit, List<Diagnostic> diagnostics) {
-
         if (unit != null) {
             PsiClass[] alltypes;
             PsiField[] allFields;
-            PsiAnnotation[] annotations;
             PsiMethod[] allMethods;
 
             alltypes = unit.getClasses();
             for (PsiClass type : alltypes) {
                 allFields = type.getFields();
                 for (PsiField field : allFields) {
-                    annotations = field.getAnnotations();
-                    for (PsiAnnotation annotation : annotations) {
-                        String matchedAnnotation = getMatchedJavaElementName(type, annotation.getQualifiedName(),
-                                SET_OF_ANNOTATIONS.toArray(new String[0]));
-                        if (matchedAnnotation != null) {
-                            validAnnotation(field, annotation, matchedAnnotation, diagnostics);
-                        }
-                    }
+                    processAnnotations(field, type, diagnostics);
                 }
                 allMethods = type.getMethods();
                 for (PsiMethod method : allMethods) {
-                    annotations = method.getAnnotations();
-                    for (PsiAnnotation annotation : annotations) {
-                        String matchedAnnotation = getMatchedJavaElementName(type, annotation.getQualifiedName(),
-                                SET_OF_ANNOTATIONS.toArray(new String[0]));
-                        if (matchedAnnotation != null) {
-                            validAnnotation(method, annotation, matchedAnnotation, diagnostics);
-                        }
-                    }
+                    processAnnotations(method, type, diagnostics);
                 }
             }
         }
+    }
 
+    private void processAnnotations(PsiJvmModifiersOwner psiModifierOwner, PsiClass type, List<Diagnostic> diagnostics) {
```

Comment:
@mrglavas , Just to confirm, the `PsiJvmModifiersOwner` class can be used for both `PsiField` and `PsiMethod`, right?